File: Java/[2 sem] 2DWorldSimulator/code/Organism.java
public abstract class Organism implements Creature {
    protected int x;
    protected int y;
    protected int strength;
    protected int initiative;
    protected int age;
    protected char symbol;
    protected String type;
    protected boolean alive;

    public Organism(int x, int y, int strength, int initiative, char symbol, int age) {
        this.x = x;
        this.y = y;
        this.strength = strength;
        this.initiative = initiative;
        this.age = age;
        this.symbol = symbol;
        this.type = "Organism";
        this.alive = true;
    }

    public void death() {
        this.alive = false;
    }

    public boolean isAlive() {
        return this.alive;
    }

    public int getX() {
        return this.x;
    }

    public int getY() {
        return this.y;
    }

    public int getStrength() {
        return this.strength;
    }

    public int getInitiative() {
        return this.initiative;
    }

    public int getAge() {
        return this.age;
    }

    public String getSymbol() {
        return Character.toString(this.symbol);
    }

    public String getType() {
        return this.type;
    }

    public void setX(int newX) {
        this.x = newX;
    }

    public void setY(int newY) {
        this.y = newY;
    }

    public void setStrength(int newStrength) {
        this.strength = newStrength;
    }

    public void setInitiative(int newInitiative) {
        this.initiative = newInitiative;
    }

    public void setAge(int newAge) {
        this.age = newAge;
    }

    public void setSymbol(char newSymbol) {
        this.symbol = newSymbol;
    }
}

File: Python/[2 sem] 2DWorldSimulator/code/Organism.py
from dataContainers import Position
import Constants


class Organism:
    def __init__(self, world, strength: int, initiative: int, position: Position, age: int = 0):
        self._world = world
        self._age = age
        self._alive = True
        self._strength = strength
        self._initiative = initiative
        self._position = position
        self._symbol = '0'

    def get_age(self) -> int:
        return self._age

    def get_alive(self) -> bool:
        return self._alive

    def get_strength(self) -> int:
        return self._strength

    def get_initiative(self) -> int:
        return self._initiative

    def get_position(self) -> Position:
        return self._position

    def get_symbol(self) -> str:
        return self._symbol

    def set_age(self, age: int):
        self._age = age

    def set_alive(self, alive: bool):
        self._alive = alive

    def set_strength(self, strength: int):
        self._strength = strength

    def set_initiative(self, initiative: int):
        self._initiative = initiative

    def set_position(self, position: Position):
        self._position = position

    def death(self):
        self._alive = False

    def action(self):
        pass

    def collision(self, organism):
        pass

    def is_cyber_sheep(self):
        return self.get_symbol() == Constants.ORGANISMS_DATA['CyberSheep']['symbol']
